servicetalk-http-utils/src/main/java/io/servicetalk/http/utils/StacklessCancellationException.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright © 2024 Apple Inc. and the ServiceTalk project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.servicetalk.http.utils;
+
+import io.servicetalk.concurrent.internal.ThrowableUtils;
+
+import java.util.concurrent.CancellationException;
+
+final class StacklessCancellationException extends CancellationException {
```

Comment:
I was not sure if it's worth making it public in a module like `concurrent-internal` and created a copy for now. Can move if someone prefers.

Reply:
Feels a bit confusing to have the same exception in two different places, especially since http-netty depends on utils?

Reply:
Actually, I changed my mind. It's worth capturing the full stack-trace if we capture the Subscriber's stack-trace instead of the netty transport stack-trace. Updated PR